src/format/command.rs

```diff
@@ -6,21 +6,29 @@ use std::str::FromStr;
 
 #[derive(Debug, PartialEq)]
 pub struct Formatted {
-    pub path: PathBuf,
+    pub source: PathBuf,
+    pub destination: FormatDestination,
     pub input: String,
     pub output: String,
 }
 
+#[derive(Debug, PartialEq)]
+pub enum FormatDestination {
+    Stdout,
+    File { path: PathBuf },
+}
+
 pub fn run(stdin: bool, check: bool, files: Vec<String>) -> Result<(), Error> {
+    let formatted;
```

Comment:
Can we assign directly to the variable please? I much prefer the style of using expressions.

```rust
let formatted = if stdin { // ....
```

Reply:
I didn't realize that was an option! I much prefer that too

Reply:
Honestly I didn't know you could not initialise a variable. I guess Rust can keep track of whether variables are initialised in order to make it safe. I still don't like it 😅

Reply:
Yeah, it bothered me too. I'm not used to C-style languages having nice flow-control constructs, so I just assumed it was the best I could do in Rust :P This language is growing on me a little bit.

Reply:
Rust is Haskell in C's clothing :)